drivers/staging/android/ion/ion_dummy_driver.c

```diff
@@ -154,3 +188,117 @@ static void __exit ion_dummy_exit(void)
 	}
 }
 __exitcall(ion_dummy_exit);
+
+#ifdef CONFIG_ION_UNMAPED
+static struct ion_platform_heap **dummy_unmaped_pheaps;
+static struct ion_heap **dummy_unmaped_heaps;
+
+static int dummy_unmaped_heaps_remove(struct platform_device *pdev)
+{
+	struct ion_platform_heap **pheap;
+	struct ion_heap **heap;
+
+	pheap = dummy_unmaped_pheaps;
+	heap = dummy_unmaped_heaps;
+	if (pheap && heap)
+		while(*pheap) {
+			ion_unmaped_heap_destroy(*heap);
+			*heap++ = NULL;
+			devm_kfree(&pdev->dev, *pheap);
+			*pheap++ = NULL;
+		}
+	devm_kfree(&pdev->dev, dummy_unmaped_pheaps);
+	devm_kfree(&pdev->dev, dummy_unmaped_heaps);
+	dummy_unmaped_pheaps = NULL;
+	dummy_unmaped_heaps = NULL;
+
+	return 0;
+}
+
+static int dummy_unmaped_heaps_probe(struct platform_device *pdev)
+{
+	const struct device_node *dt_node = pdev->dev.of_node;
+	struct device_node *np;
+	struct ion_platform_heap **pheap;
+	struct ion_heap **heap;
+	size_t num_heaps = 0;
+	int rc;
+
+	if (dummy_unmaped_heaps) {
+		// TODO: need to support several nodes ?
```

Comment:
I don't think that is necessary, not as of now at least. One should be enough for now I think.